src/Toolchain/ToolchainProvider.ts

```diff
@@ -182,15 +192,51 @@ export class ToolchainProvider implements vscode.TreeDataProvider<BaseNode> {
     }
 
     gToolchainEnvMap[backendName].uninstall(tnode.toolchain).then(() => {
+      if (DefaultToolchain.getInstance().equal(tnode.toolchain)) {
+        Logger.info(this.tag, 'Default toolchain is cancelled.');
+        DefaultToolchain.getInstance().unset();
+      }
       this.refresh();
     });
   }
 
-  run(_cfg: string) {
-    throw Error('Not implemented yet');
+  run(cfg: string, backend?: string, toolchain?: Toolchain) {
+    const activeToolchainEnv =
+        backend ? gToolchainEnvMap[backend] : DefaultToolchain.getInstance().getToolchainEnv();
+    const activeToolchain = toolchain ? toolchain : DefaultToolchain.getInstance().getToolchain();
```

Comment:
When we have two backends (`foo` and `bar`) and default toolchain is `bar 1.0`,
I am afraid that, e.g., `run(a.cfg, foo)` where param `toolchain === undefined` could lead mismatch env and toolchain (`activeToochainEnv === gToolchainEnvMap[foo]` and `activeToolchain === bar 1.0`).

Reply:
I already handled the mismatch case. 

```
    if (!activeToolchainEnv?.listInstalled().find(value => JSON.stringify(value.info) === JSON.stringify(activeToolchain.info))) {
      this.error('Invalid Toolchain');
      return;
    }
```

Does this code resolve your concern?

Reply:
I am afraid that `listInstalled()` might be somewhat heavy for this check.,

If there is no fancy way,
how about
- ` run(cfg: string, backend?: string, toolchain?: Toolchain)` ==>  `run(cfg: string)` if we don't have serious requirements with `backend` and `toolchain` params
- or adding `//TODO ...` T.T

Reply:
I thought there would be a UX where the user directly selects and calls the toolchain without using the default toolchain. For that reason, it was designed to receive more arguments. But.. I will fix it.